File: src/main/java/org/interledger/everledger/AuthInfo.java
package org.interledger.everledger;


public class AuthInfo {
    final public String id;
    final public String name;
    final public String pass;
    final public String roll; // TODO:(1) Create ENUM 

    public static final AuthInfo ANONYMOUS = new AuthInfo("","","", "none");
    
    public AuthInfo(String id, String name, String pass, String roll) {
        this.id = id;
        this.name = name;
        this.pass = pass;
        this.roll = roll;
    }

    public String getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getRoll() {
        return roll;
    }

    public boolean isAdmin() {
        return "admin".equals(roll);
    }

    public boolean isConnector() {
        return "connector".equals(roll);
    }

    @Override
    public String toString() {
        return pass;
    }
    
    

}

File: src/main/java/com/everis/everledger/AuthInfo.kt
package com.everis.everledger

enum class AccessRoll {
    ADMIN,
    CONNECTOR,
    USER,
    NONE

}

// TODO:(1) Change pass by hashOfPass + salt + ...
data class AuthInfo( val id: String, val login: String, val pass: String, val roll: AccessRoll ) {

    val isAdmin: Boolean
        get() = roll == AccessRoll.ADMIN

    val isConnector: Boolean
        get() = roll == AccessRoll.ADMIN

    override fun toString() = id

    companion object {
        val ANONYMOUS = AuthInfo("", "", "", AccessRoll.NONE)
    }
}